addon/components/o-s-s/toggle-buttons.ts

```diff
@@ -0,0 +1,43 @@
+import { assert } from '@ember/debug';
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export type toggle = {
+  value: string;
+  label: string;
+};
+
+interface OSSToggleButtonsArgs {
+  toggles: toggle[];
+  selectedToggle?: string;
+  onSelection(selectedToggle: string): void;
+}
+
+export default class OSSToggleButtons extends Component<OSSToggleButtonsArgs> {
+  constructor(owner: unknown, args: OSSToggleButtonsArgs) {
+    super(owner, args);
+
+    assert('[component][OSS::ToggleButtons] The @toggles parameter of type object is mandatory', args.toggles);
+
+    assert(
+      '[component][OSS::ToggleButtons] The @onSelection parameter of type function is mandatory',
+      typeof args.onSelection === 'function'
+    );
+  }
```

Comment:
Define `@selectedToggle` as required to avoid loop with computed properties when trigger `@onSelection`

```js
Uncaught Error: Assertion Failed: You attempted to update `selectedToggle` on `<ApplicationController:ember134>`, but it had already been used previously in the same computation.  Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, and performance issues, and is not supported.
```